libminifi/test/unit/StringViewTests.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <vector>
+#include <tuple>
+#include "utils/StringView.h"
+#include "utils/StringViewUtils.h"
+#include "../TestBase.h"
+
+using utils::StringView;
+using utils::StringViewUtils;
+
+TEST_CASE("Comparision") {
+  std::string a = "a";
+  std::string b = "b";
+
+  REQUIRE(StringView(a) == "a");
+  REQUIRE(StringView(a) != "b");
+  REQUIRE(StringView("a") == "a");
+  REQUIRE(StringView("abcd") != "a");
+  REQUIRE(StringView(a) == a);
+  REQUIRE(StringView(a) != b);
+  REQUIRE(StringView(a) == StringView(a));
+  REQUIRE(StringView(a) != StringView(b));
+  REQUIRE(!(StringView(a) == StringView(b)));
+}
+
+TEST_CASE("trimLeft") {
+  std::vector<std::pair<std::string, std::string>> cases{
+      {"", ""},
+      {"abcd", "abcd"},
+      {" abc", "abc"},
+      {"  abc", "abc"},
+      {" ", ""},
+      {"abc ", "abc "}
```

Comment:
throw in a few more cases where the whitespace is \t, \r or \n instead of spaces?

Reply:
added